client/dashboard/components/guided-tour/context.tsx

```diff
@@ -75,16 +75,30 @@ export const GuidedTourContextProvider = ( {
 	const endTour = useCallback( () => {
 		// Save the dismissed timestamp so we can show the new steps that are added after it in the future.
 		updateCompletedTimestamp( new Date().toISOString() );
-		recordTracksEvent( 'calypso_dashboard_end_tour', { tour_id: tourId } );
-	}, [ tourId, updateCompletedTimestamp, recordTracksEvent ] );
+		recordTracksEvent( 'calypso_dashboard_end_tour', {
+			tour_id: tourId,
+			is_completed: currentStep === guidedTours.length,
+		} );
+	}, [ tourId, updateCompletedTimestamp, recordTracksEvent, currentStep, guidedTours.length ] );
 
 	const previousStep = useCallback( () => {
-		setCurrentStep( ( step ) => Math.max( step - 1, 0 ) );
-	}, [] );
+		setCurrentStep( ( step ) => {
+			const nextStep = Math.max( step - 1, 0 );
+			recordTracksEvent( 'calypso_dashboard_previous_step_tour', { to: nextStep } );
```

Comment:
Maybe `calypso_dashboard_tour_previous_step` instead?

(Also similarly for next_step)